sources/core/Stride.Core.Yaml/Serialization/DefaultObjectFactory.cs

```diff
@@ -119,11 +120,12 @@ public object Create(Type type)
                 }
             }
 
-            return null;
+            throw new InstanceCreationException($"Failed to create instance of type '{type}', type does not implement a parameterless constructor.");
```

Comment:
Nit: a type *has* or *doesn't have* a parameterless constructor. It doesn't really implement it, the same way an interface or abstract method can be.

```suggestion
            throw new InstanceCreationException($"Failed to create instance of type '{type}', type does not have a parameterless constructor.");
```

Reply:
done